src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncidentMappingResourceIntTest.kt

```diff
@@ -0,0 +1,545 @@
+            mappingType = mappingType,
+          ),
+        ),
+      )
+      .exchange()
+      .expectStatus().isCreated
+  }
+
+  @DisplayName("POST /mapping/incidents")
+  @Nested
+  inner class CreateIncidentMappingTest {
+
+    @AfterEach
+    internal fun deleteData() = runBlocking {
+      repository.deleteAll()
+    }
+
+    @Test
+    fun `access forbidden when no authority`() {
+      webTestClient.post().uri("/mapping/incidents")
+        .body(BodyInserters.fromValue(createIncidentMapping()))
+        .exchange()
+        .expectStatus().isUnauthorized
+    }
+
+    @Test
+    fun `access forbidden when no role`() {
+      webTestClient.post().uri("/mapping/incidents")
+        .headers(setAuthorisation(roles = listOf()))
+        .body(BodyInserters.fromValue(createIncidentMapping()))
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `create forbidden with wrong role`() {
+      webTestClient.post().uri("/mapping/incidents")
+        .headers(setAuthorisation(roles = listOf("ROLE_BANANAS")))
+        .body(BodyInserters.fromValue(createIncidentMapping()))
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `create when mapping for incident id already exists for another mapping`() {
+      postcreateIncidentMappingRequest()
+
+      val responseBody = webTestClient.post().uri("/mapping/incidents")
+        .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_INCIDENTS")))
+        .contentType(MediaType.APPLICATION_JSON)
+        .body(BodyInserters.fromValue(createIncidentMapping().copy(nomisIncidentId = 21)))
+        .exchange()
+        .expectStatus().isEqualTo(409)
+        .expectBody(object : ParameterizedTypeReference<DuplicateMappingErrorResponse<IncidentMappingDto>>() {})
+        .returnResult().responseBody
+
+      with(responseBody!!) {
+        assertThat(userMessage).contains("Conflict: Incident mapping already exists. \nExisting mapping: IncidentMappingDto(nomisIncidentId=1234, incidentId=4444, label=2022-01-01, mappingType=NOMIS_CREATED")
+        assertThat(userMessage).contains("Duplicate mapping: IncidentMappingDto(nomisIncidentId=21, incidentId=4444, label=2022-01-01, mappingType=NOMIS_CREATED, whenCreated=null)")
+        assertThat(errorCode).isEqualTo(1409)
+      }
+
+      val existingIncident = responseBody.moreInfo?.existing!!
+      with(existingIncident) {
+        assertThat(incidentId).isEqualTo("4444")
+        assertThat(nomisIncidentId).isEqualTo(1234)
+        assertThat(mappingType).isEqualTo("NOMIS_CREATED")
+      }
+
+      val duplicateIncident = responseBody.moreInfo?.duplicate!!
+      with(duplicateIncident) {
+        assertThat(incidentId).isEqualTo("4444")
+        assertThat(nomisIncidentId).isEqualTo(21)
+        assertThat(mappingType).isEqualTo("NOMIS_CREATED")
+      }
+    }
+
+    @Test
+    internal fun `create mapping does not error when the same mapping already exists for the same adjustment`() {
```

Comment:
copy n paste typo "adjustment"

Reply:
will fix in next PR - when that eventually comes along